dawn/src/dawn/Optimizer/PassValidation.cpp

```diff
@@ -0,0 +1,67 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassValidation.h"
+
+namespace dawn {
+
+PassValidation::PassValidation(OptimizerContext& context) : Pass(context, "PassValidation") {}
+
+bool PassValidation::run(const std::shared_ptr<iir::StencilInstantiation>& instantiation) {
+  return run(instantiation, "");
+}
+
+bool PassValidation::run(const std::shared_ptr<iir::StencilInstantiation>& instantiation,
+                         const std::string& description) {
+  IntegrityChecker checker(instantiation.get());
+  checker.run();
```

Comment:
This throws, we should catch it

Reply:
Exceptions are currently caught in the PassManager and added to the diagnostics.